### ISSUE TITLE
extension lacking fundamental automatization features and relies heavily on user-input and "editor.fontFamily"

### ISSUE THREAD

**Issue by NinjaZMY** [Zev18/font-randomizer#3]
1. Doesn't randomize font 
2.  every time the extension runs it makes it own imaginative list of font in my case monospace mentioned twice
 ( loading from a font list itself is wrong, as other extensions would load system fonts which is way more sophisticated, more accurate, more automatic, and wouldn't include misinformation as your list can be a useless list whether fonts don't exist or aren't installed in the system itself)

**Comment by Zev18** [comment]
1. Did you try restarting vscode?
2. What do you mean "it's own imaginative list" and "monospace mentioned twice?" Can you provide a screenshot?

**Comment by NinjaZMY** [comment]
# ps: I have removed the "editor.fontFamily" line from my settings.js ... yet
![image](https://github.com/user-attachments/assets/89588332-a96a-4896-8663-13f41af3fd41)

>on extension install, it creates ` "editor.fontFamily": "monospace, monospace" ` 
> 
>
>  < which I referred as an imaginative list > 

on` window.reload` > **`monspace`** is than picked by default which is logical since you have to randomize from the list 

by the way I'm using other font extensions and each have functioning font 

> < not an imaginative list like `"monospace,monospace"` >

### monspace font functions but why would I need such an almost empty list, if it doesn't list functional font and is dysfunctional spamming monospace on each time the window is reloaded

**Comment by Zev18** [comment]
What does your `fontRandomizer.fontList` setting look like?

**Comment by Zev18** [comment]
Also could you please not use insults like "useless extension should be deleted from marketplace else make something that works instead of writing low effort code?" I worked really hard on this and it hurts my feelings :(

**Comment by NinjaZMY** [comment]
1) `fontRandomizer.fontList` wasn't there at first 

2) adding it with actual functional font will function but still lacks automation 

- [ ] Automate font detection and dynamic list generation 
   - [ ] #2  
- [ ] [required for an independent extension, so that overrides won't interfere with extension functionality ] make the extension work so that you won't have to edit "editor.fontFamily"


![image](https://github.com/user-attachments/assets/b95bd799-7c5b-4d44-8c11-f879b261f7d6)

**Comment by NinjaZMY** [comment]
# I'm sorry for hurting your feelings, I'm really sorry for saying hurtful things I didn't mean it like that 
## Seeing your profile I really liked your work to be honest, it was really good 
However, this extension compared to the existing solutions is far behind, the way it wasn't fully automatisé / automatized

**Comment by Zev18** [comment]
Ok. I just pushed an update that should fix all the issues. Let me know if you have any more problems. If not, I would really appreciate it if you changed your review on the vscode marketplace to 5 stars

**Comment by NinjaZMY** [comment]
> 1. `fontRandomizer.fontList` wasn't there at first
> 2. adding it with actual functional font will function but still lacks automation
> 
> * [ ]  Automate font detection and dynamic list generation
>   
>   * [ ]  [Read current font settings and add them to fontList once installed #2](https://github.com/Zev18/font-randomizer/issues/2)
> * [ ]  [required for an independent extension, so that overrides won't interfere with extension functionality ] make the extension work so that you won't have to edit "editor.fontFamily"
> 
> ![image](https://private-user-images.githubusercontent.com/45049872/356907054-b95bd799-7c5b-4d44-8c11-f879b261f7d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM2ODk4MjAsIm5iZiI6MTcyMzY4OTUyMCwicGF0aCI6Ii80NTA0OTg3Mi8zNTY5MDcwNTQtYjk1YmQ3OTktN2M1Yi00ZDQ0LThjMTEtZjg3OWIyNjFmN2Q2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDAyMzg0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1OGVjYjAxNjFmYTk4YzYxOTA1YTdmYTFhMzE5NzRmZTg3MmJhNDU3ZmQ0MDFkZTYyOTRmNmViNDBlODRlZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tYwjHG_kwtqVdFCAmna9No0FSetwkPOxAPXR0Trs-5Q)

# you don't seem to understand the gravity of your issues or the lacking functionalities or maybe your testing skills aren't topnotch
### Anyway, the fixes you've pushed don't fix the problems I mentioned above 
# 
problems: 

- [x] zero automatisations, user is obliged to not only add extension settings but also feed it with manual input 
- [x] extension heavily relies on a global parameter from settings.json which can be easily overridden or deleted 
- [x] extension isn't loading system fonts into its fontList making the list potentially prone to invalid input, which confirms that the fontList/extension isn't efficient
- [x] fontList cannot handle exceptions or invalid input

**Comment by Zev18** [comment]
I don't think you understand the point of this extension. This extension isn't supposed to automatically select from every font you have. It's supposed to give you the ability to manually create a list, and then selects from that list. I don't know what other "automations" I could possibly add other than creating the initial font list from the `editor.fontFamily` setting. Also with my new "add font" command now you don't have to worry about invalid input.
